apps/users.py

```diff
@@ -106,6 +111,71 @@ def logout():
     return redirect(url_for('base.main'))
 
 
+class SignupForm(Form):
+    email = EmailField('Email', [Email(), Required()])
+    name = StringField('Name', [Required()])
+    allow_promo = BooleanField('Send me occasional emails about future EMF events')
+    signup = SubmitField('Sign up')
+
+    def validate_email(form, field):
+        if User.does_user_exist(field.data):
+            field.was_duplicate = True
+
+            msg = Markup(
+                render_template_string('Account already exists. '
```

Comment:
will these strings be auto-concatenated?

Reply:
yep